### PR TITLE
Strong 'boost' parameter

### PR DESCRIPTION
**Pull request by allevo**
Add strong type to `boost` parameter

### CODE REVIEW

File: packages/orama/src/types.ts [hunk 1]
```diff
@@ -236,7 +239,7 @@ export type SearchParams<T extends AnyOrama, ResultDocument = TypedDocument<T>>
   /**
    * The properties of the document to search in.
    */
-  properties?: '*' | FlattenSchemaProperty<T>[]
+  properties?: '*' | FlattenSchemaProperties<T>
```
Comment (micheleriva) [inline packages/orama/src/types.ts:242]: Why converting this to plural form?

Comment (allevo) [reply packages/orama/src/types.ts:242]: Because it is an array of FlattenSchemaProperty.
Do you prefer another convention?

Comment (micheleriva) [reply packages/orama/src/types.ts:242]: Yes by convention it is better to use singular and convert them to arrays when used in lists

Comment (allevo) [reply packages/orama/src/types.ts]: done!

Comment (allevo) [reply packages/orama/src/types.ts]: done!

**Review by micheleriva — APPROVED** [review]

**Pull request merged**